kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/grid/model/DMNGridColumn.java

```
@@ -51,6 +54,24 @@ public DMNGridColumn(final List<HeaderMetaData> headerMetaData,
         this.gridWidget = gridWidget;
     }
 
+    protected GridCell<T> assertCell(final GridCell<T> cell) {
+        if (cell != null) {
+            return cell;
+        }
+        return new BaseGridCell<>(makeDefaultCellValue());
+    }
+
+    protected GridCellValue<T> assertCellValue(final GridCellValue<T> cellValue) {
+        if (cellValue != null) {
+            return cellValue;
+        }
+        return makeDefaultCellValue();
+    }
+
+    protected GridCellValue<T> makeDefaultCellValue() {
+        return null;
```

Question here, wouldn't be better to throw an exception or make the method abstract to force user override appropriately in `DMNGridColumn` implemetations?

---

I've moved the assertions to a new sub-class ```DMNSimpleGridColumn``` that is the base for all "simple" columns that use ```TextBox``` or ```TextArea```. The only columns not extending this are ```ExpressionEditorColumn```, ```UndefinedExpressionColumn``` (both of which provide a custom widget for editing the cell value) and ```function.supplementary.NameColumn``` that is not editable. There is now no need to either have all columns implement ```makeDefaultCellValue()``` or throw an exception.